lib/rubocop/cop/layout/update_info_on_new_line.rb

```diff
@@ -0,0 +1,62 @@
+module RuboCop
+  module Cop
+    module Layout
+      class UpdateInfoOnNewLine < Cop
+        include Alignment
+
+        MSG = "%<name>s should start on its own line"
+
+        def_node_matcher :find_update_info_node, <<~PATTERN
+          (def :initialize _args (begin (super $(send nil? :update_info (lvar :info) (hash ...))) ...))
+        PATTERN
+
+        def_node_matcher :find_nested_update_info_node, <<~PATTERN
+          (def :initialize _args (super $(send nil? :update_info (lvar :info) (hash ...)) ...))
+        PATTERN
```

Comment:
Declarative AST matching pattern:

https://docs.rubocop.org/en/latest/node_pattern/

Not a fan of the duplicated node + nested node definition - which is required to capture both a single call to `super` as well as `super` and sibling elements, i.e. `register_options` and friends - but using backticks for descendants didn't seem to work in this scenario. Thankfully there's tests, and the specifics of this rule's implementation don't matter too much.